ophyd/device.py

```diff
@@ -184,12 +189,48 @@ def __get__(self, instance, owner):
             return self
 
         if self.attr not in instance._signals:
-            instance._signals[self.attr] = self.create_component(instance)
+            cpt_inst = self.create_component(instance)
+            instance._signals[self.attr] = cpt_inst
+
+            for event_type, functions in self._subscriptions.items():
+                for func in functions:
+                    cpt_inst.subscribe(types.MethodType(func, instance),
+                                       event_type=event_type,
+                                       run=cpt_inst.connected)
```

Comment:
So we will run the callback if the `cp_inst` is `lazy`?

Reply:
How this works (admittedly somewhat confusingly) is as follows:
Callback data is cached on the ophydobject level. If you subscribe for the first time, you have the option of using that cached information to run the callback immediately. If you don't do that, then the next time you'll see an update is when the subscription fires again for all other subscribers.

This specific line works as follows: if it's already connected, run the subscription, otherwise wait until connected as we're likely to get the first valid subscription callback from that.